api/src/main/java/ai/djl/util/ZipUtils.java

```diff
@@ -121,14 +118,16 @@ private static void addToZip(Path root, Path file, ZipOutputStream zos) throws I
         }
     }
 
-    static String removeLeadingFileSeparator(String name) {
-        int index = 0;
-        for (; index < name.length(); index++) {
-            if (name.charAt(index) != File.separatorChar) {
-                break;
-            }
+    static String validateArchiveEntry(String name, Path destination) throws IOException {
+        Path expectedOutputPath = destination.resolve(name).normalize();
+        if (!expectedOutputPath.startsWith(destination.normalize())) {
+            throw new IOException(
+                    "Bad archive entry "
+                            + name
+                            + ". Attempted write outside destination "
+                            + destination);
         }
-        return name.substring(index);
+        return name;
```

Comment:
Why return the input variable as is? Maybe this method should be `getAbsolutePath` and the method is expected to throw if the path is invalid. 
We are using the output in `vis.validate`. Do we need that after this change?

Reply:
Agree, this method can be `void`

Reply:
I think we still need `vis.validate` since that method is doing some deeper validation specific to zips. It's validating the entries in the header match what is in the archive